packages/tiled/lib/src/common/property.dart

```diff
@@ -12,35 +12,213 @@ part of tiled;
+        );
+
+      case PropertyType.bool:
+        return BoolProperty(
+          name: name,
+          value: parser.getBool('value', defaults: false),
+        );
+
+      case PropertyType.float:
+        return FloatProperty(
+          name: name,
+          value: parser.getDouble('value', defaults: 0),
+        );
+
+      case PropertyType.int:
+        return IntProperty(
+          name: name,
+          value: parser.getInt('value', defaults: 0),
+        );
+
+      case PropertyType.file:
+        return FileProperty(
+          name: name,
+          value: parser.getString('value', defaults: '.'),
+        );
+
+      case PropertyType.string:
+        final value = parser.formatSpecificParsing((json) {
+          return json.getString('value', defaults: '');
+        }, (xml) {
+          final attrString = parser.getStringOrNull('value');
+          if (attrString != null) {
+            return attrString;
+          } else {
+            // In tmx files, multi-line text property values can be stored
+            // inside the <property> node itself instead of in the 'value'
+            // attribute
+            return xml.element.innerText;
+          }
+        });
+
+        return StringProperty(
+          name: name,
+          value: value,
         );
+    }
+  }
+}
+
+/// A wrapper for a Tiled property set
+///
+/// Accessing an int value
+/// ```dart
+/// properties.get<int>('foo') == 3
+/// ```
+///
+/// Accessing an int property:
+/// ```dart
+/// properties.getProp<IntProperty>('foo') ==
+///     IntProperty(name: 'foo', value: 3);
+/// ```
+class CustomProperties extends Iterable<Property<Object>> {
+  static const empty = CustomProperties({});
+
+  /// The properties, indexed by name
+  final Map<String, Property<Object>> byName;
+
+  const CustomProperties(this.byName);
+
+  /// Get a property value by its name.
+  ///
+  /// [T] must be the type of the value, which depends on the property type.
+  /// The following Tiled properties map to the follow Dart types:
+  ///  - int property -> int
+  ///  - float property -> double
+  ///  - boolean property -> bool
+  ///  - string property -> string
+  ///  - color property -> ui.Color
+  ///  - file property -> string (path)
+  ///  - object property -> int (ID)
+  T? getValue<T>(String name) {
+    return getProperty(name)?.value as T?;
+  }
+
+  /// Get a typed property by its name
+  T? getProperty<T extends Property<Object>>(String name) {
+    return byName[name] as T?;
+  }
+
+  /// Get a property by its name
+  Property<Object>? operator [](String name) {
+    return byName[name];
+  }
+
+  /// Returns whether or not a property with [name] exists.
+  bool has(String name) {
+    return byName.containsKey(name);
+  }
+
+  @override
+  Iterator<Property<Object>> get iterator => byName.values.iterator;
+}
+
+/// [value] is the ID of the object
+class ObjectProperty extends Property<int> {
```

Comment:
```suggestion
class ObjectProperty extends Property<Object> {
```

Reply:
This is correct actually, the implementation at this stage only give you access to the Tiled object's integer ID